externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/spark/sql/kyuubi/SparkDatasetHelper.scala

```diff
@@ -290,4 +290,13 @@ object SparkDatasetHelper extends Logging {
     nodeName == "org.apache.spark.sql.execution.command.ExecutedCommandExec" ||
     nodeName == "org.apache.spark.sql.execution.CommandResultExec"
   }
+
+  /** SPARK-47270: Returns a optimized plan for CommandResult, convert to `LocalRelation`. */
+  def commandResultOptimized[T](dataset: Dataset[T]): Dataset[T] = {
```

Comment:
I'm -0 on this change, I would consider it a Spark side issue. Spark’s master branch is undergoing a major refactoring, I'm worried about accessing Spark's non-public API in the engine's "core code path".

For users who want to avoid triggering executor launch, they can either patch their spark or set init SQL as something like: `SET spark.app.id`

Reply:
Ok. I will cherry-pick the pr into our managed spark 3.5.